### PR TITLE
Unbound analysis for drop table/drop blob table

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by mkleen** [comment]
retest this please 

Running: Test class io.crate.planner.consumer.GroupByPlannerTest

After 20s process did not stop

**Comment by mkleen** [comment]
Is this all i had to do, or I am missing the point here ?

File: sql-parser/src/main/java/io/crate/sql/tree/DropTable.java [hunk 1]
```diff
@@ -47,6 +49,10 @@ public Table table() {
         return visitor.visitDropTable(this, context);
     }
 
+    public <U> DropTable<U> map(Function<? super T, ? extends U> mapper) {
```
Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/tree/DropTable.java]: is this used anywhere?

File: sql-parser/src/main/java/io/crate/sql/tree/DropBlobTable.java [hunk 2]
```diff
@@ -43,6 +45,10 @@ public Table table() {
         return table;
     }
 
+    public <U> DropTable<U> map(Function<? super T, ? extends U> mapper) {
```
Comment (seut) [inline sql-parser/src/main/java/io/crate/sql/tree/DropBlobTable.java]: also this is not used I think and can be removed

File: sql/src/main/java/io/crate/analyze/UnboundAnalyzer.java [hunk 3]
```diff
@@ -259,8 +266,8 @@ public AnalyzedStatement visitUpdate(Update update, Analysis analysis) {
         }
 
         @Override
-        protected AnalyzedStatement visitExplain(Explain node, Analysis context) {
-            return explainStatementAnalyzer.analyze(node, context);
+        protected AnalyzedStatement visitExplain(Explain node, Analysis analysis) {
+            return explainStatementAnalyzer.analyze(node, analysis);
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/UnboundAnalyzer.java]: unrelated changes should be avoided.

File: sql/src/main/java/io/crate/analyze/UnboundAnalyzer.java [hunk 4]
```diff
@@ -271,5 +278,14 @@ public AnalyzedStatement visitOptimizeStatement(OptimizeStatement<?> node, Analy
                 context.transactionContext()
             );
         }
+
+        public AnalyzedDropTable visitDropTable(DropTable<?> node, Analysis context) {
+            return dropTableAnalyzer.analyze(node,context.sessionContext());
+        }
+
+        public AnalyzedStatement visitDropBlobTable(DropBlobTable<?> node, Analysis analysis) {
+            return visitNode(node, analysis);
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/UnboundAnalyzer.java]: should be `dropTableAnalyzer.analyze(node, context.sessionContext());` I guess..

**Comment by mkleen** [comment]
retest this please

**Comment by mkleen** [comment]
retest this please 

Running: Test class io.crate.rest.action.RestActionReceiversTest



> Task :sql:test FAILED



> Task :enterprise:lang-js:test

Running: Gradle Test Run :enterprise:lang-js:test

Running: Gradle Test Executor 27

Running: Test class io.crate.operation.language.JavaScriptUDFIntegrationTest

Running: Gradle Test Executor 28

Running: Test class io.crate.operation.language.JavascriptUserDefinedFunctionTest



The message received from the daemon indicates that the daemon has disappeared.

Build request sent: Build{id=06cf8cd7-3e5b-4b5a-853b-e03809c517e1, currentDir=/var/lib/jenkins/workspace/CrateDB/crate_on_pr}

Attempting to read last messages from the daemon log...

Daemon pid: 8113

  log file: /var/lib/jenkins/.gradle/daemon/6.0-20191006230018+0000/daemon-8113.out.log

----- Last  20 lines from daemon log file - daemon-8113.out.log -----

	at org.gradle.process.internal.DefaultExecHandle.failed(DefaultExecHandle.java:372)

	at org.gradle.process.internal.ExecHandleRunner.run(ExecHandleRunner.java:87)

	at org.gradle.internal.operations.CurrentBuildOperationPreservingRunnable.run(CurrentBuildOperationPreservingRunnable.java:42)

	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)

	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)

	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)

	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)

	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)

	at java.base/java.lang.Thread.run(Thread.java:834)

Caused by: java.lang.IllegalStateException: Shutdown in progress

**Comment by seut** [comment]
Is this still a draft?

**Comment by mkleen** [comment]
> Is this still a draft?

oh my bad, yes.
